lab_2/main.py

```diff
@@ -27,32 +27,99 @@ def read_ecg_rr_file(file_path: Path):
         rr_signal.append(numeric_value)
     return rr_signal
 
+# def test_input_of_float(input:list):
+#     if type(input) != list or len(input)==0:
+#         return None
+#     for i in input:
+#         if type(i) != float:
+#             return None
+
+def calculating_rrd (signal: list):
+    rrd=[]
+    for i in range(0,len(signal [1:])):
```

Comment:
creating intermediate copy of a list that takes memory although we do not need it